src/Symfony/Component/ExpressionLanguage/ExpressionLanguage.php

```diff
@@ -32,12 +34,21 @@ class ExpressionLanguage
     protected $functions = array();
 
     /**
-     * @param ParserCacheInterface                  $cache
+     * @param CacheItemPoolInterface                $cache
      * @param ExpressionFunctionProviderInterface[] $providers
      */
-    public function __construct(ParserCacheInterface $cache = null, array $providers = array())
+    public function __construct($cache = null, array $providers = array())
     {
-        $this->cache = $cache ?: new ArrayParserCache();
+        if (null !== $cache) {
+            if ($cache instanceof ParserCacheInterface) {
+                @trigger_error(sprintf('Passing the %s as constructor argument for %s is deprecated as of 3.2 and will be removed in 4.0. Pass an instance of the %s interface instead.', ParserCacheInterface::class, self::class, CacheItemPoolInterface::class), E_USER_DEPRECATED);
```

Comment:
`Passing an instance of %s as constructor argument for %s is deprecated as of 3.2 and will be removed in 4.0. Pass an instance of %s instead.`